src/sparseml/transformers/sparsification/question_answering.py

```diff
@@ -210,7 +210,7 @@ def __init__(
     def _remove_unused_columns(
         self, dataset: "datasets.Dataset", description: Optional[str] = None
     ):
-        if self._signature_columns is None and self.teacher is not None:
+        if self._signature_columns is None and self.teacher is not None and self.teacher not in ("disable", "self"):
```

Comment:
Can you put in the same fix to a similar function inside sparsification/trainer.py?